src/analysis/calculus/times_cont_diff.lean

```diff
@@ -2178,6 +2222,91 @@ begin
   exact times_cont_diff_at.prod_map hf hg
 end
 
+/-! ### Inversion in a complete normed algebra
+
+This section is dedicated to proving that, in a complete normed algebra, the inversion operation is
+smooth.  The proof is by induction, bootstrapping using an identity involving the derivative of
+inversion as a bilinear map of inversion itself.  Constructing this bilinear map requires, in turn,
+several auxiliary lemmas. -/
+
+section algebra_inverse
+variables (𝕜)
+variables (R :Type*) [normed_ring R] [normed_algebra 𝕜 R] [complete_space R]
+open normed_ring continuous_linear_map ring
+
+/-- The function `R × R → (R →ₗ[𝕜] R)` which sends `(v, w)` to the operation of simultaneous
+left-multiplication by `v` and right-multiplication by `w`.  This is an auxiliary definition;
+in general one should use the more structured `left_right_mul`. -/
+private def left_right_mul_aux (vw: R × R) : R →ₗ[𝕜] R :=
+- ((continuous_linear_map.lmul_right 𝕜 R vw.2).comp (continuous_linear_map.lmul_left 𝕜 R vw.1))
+
+private lemma left_right_mul_aux_bound (vw : R × R) :
```

Comment:
All this part on the continuous version of left-right multiplication is not really specific to higher differentiability. Could you move it next to the definitions of `continuous_linear_map.lmul_left` and `continuous_linear_map.lmul_right`, and maybe name it `continuous_linear_map.lmul_left_right`?